src/getPubkey.c

```diff
@@ -7,7 +7,7 @@
 static uint8_t publicKey[PUBKEY_LENGTH];
 static char publicKeyStr[BASE58_PUBKEY_LENGTH];
 
-int read_derivation_path(const uint8_t *dataBuffer, size_t size, uint32_t *derivationPath) {
+uint32_t read_derivation_path(const uint8_t *dataBuffer, size_t size, uint32_t *derivationPath) {
```

Comment:
Why not `size_t`?

Reply:
I have a slight bias into using clear data sizes over platform dependent sizes.
I'll change if you prefer the other way around?

Reply:
I understand, however in this case `size_t` and `uint32_t` are mixed up in the code (this function returns a `len` variable which is a `size_t`), resulting in potential issues if the code was to run on a platform where the two types were not aliases.

In fact, it may not be that clear that `size_t` and `uint32_t` are aliases in this case: in [the main, an `int` representing a size was replaced with  `uint16_t`](https://github.com/LedgerHQ/app-solana/pull/35/files#diff-e0cf5b28d9b6b600f0af2bc78e8fd30ec675fd731a5da86f0c4283ffc0e40176R57).

This is a bit confusing at a glance.

`size_t` is platform dependent, but it's consistent.